File: app/src/main/java/com/witype/Dragger/mvp/activity/SplashActivity.java
package com.witype.Dragger.mvp.activity;

import android.content.Intent;

import com.witype.Dragger.R;
import com.witype.Dragger.mvp.BaseActivity;
import com.witype.mvp.di.component.AppComponent;

import java.util.concurrent.TimeUnit;

import androidx.annotation.NonNull;
import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

public class SplashActivity extends BaseActivity {

    private Disposable subscribe;

    @Override
    public int getResId() {
        return R.layout.activity_splash;
    }

    @Override
    protected void setupComponent(@NonNull AppComponent appComponent) {

    }

    @Override
    public void initView() {
        super.initView();
        subscribe = Observable.timer(900, TimeUnit.MICROSECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<Long>() {
                    @Override
                    public void accept(Long aLong) throws Exception {
                        toMain();
                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        throwable.printStackTrace();
                    }
                });
    }

    private void toMain() {
        startActivity(new Intent(this,MainActivity.class));
        finish();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (subscribe != null) {
            subscribe.dispose();
        }
    }
}

File: kotlindemo/src/main/java/com/witype/kotlindemo/mvp/ui/SplashActivity.kt
package com.witype.kotlindemo.mvp.ui

import android.content.Intent
import com.witype.kotlindemo.R
import com.witype.kotlindemo.mvp.presenter.IBasePresenter
import com.witype.mvp.di.component.AppComponent
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import java.util.concurrent.TimeUnit

class SplashActivity : BaseActivity<IBasePresenter>() {

    private var subscribe: Disposable? = null

    override val resId: Int
        get() = R.layout.activity_splash

    override fun setupComponent(appComponent: AppComponent) {}

    override fun initView() {
        super.initView()
        subscribe = Observable.timer(900, TimeUnit.MICROSECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe({ toMain() }) { throwable -> throwable.printStackTrace() }
    }

    private fun toMain() {
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }

    override fun onDestroy() {
        super.onDestroy()
        if (subscribe != null) {
            subscribe!!.dispose()
        }
    }
}